src/lib/bootloader/systemdboot.rb

```diff
@@ -0,0 +1,304 @@
+# frozen_string_literal: true
+
+require "fileutils"
+require "yast"
+require "bootloader/sysconfig"
+require "bootloader/cpu_mitigations"
+
+Yast.import "Report"
+Yast.import "Arch"
+Yast.import "ProductFeatures"
+Yast.import "BootStorage"
+Yast.import "Stage"
+
+module Bootloader
+  # Represents systemd bootloader with efi target
+  # rubocop:disable Metrics/ClassLength
+  class SystemdBoot < BootloaderBase
+    include Yast::Logger
+    include Yast::I18n
+
+    # @!attribute menue_timeout
+    #   @return [Integer] menue timeout
+    attr_accessor :menue_timeout
+
+    # @!attribute secure_boot
+    #   @return [Boolean] current secure boot setting
+    attr_accessor :secure_boot
+
+    def initialize
+      super
+
+      textdomain "bootloader"
+      self.menue_timeout = Yast::ProductFeatures.GetIntegerFeature("globals", "boot_timeout").to_i
+    end
+
+    def merge(other)
+      log.info "merging with system: timeout=#{other.menue_timeout} " \
+               "secure_boot=#{other.secure_boot}"
+      super
+      self.menue_timeout = other.menue_timeout unless other.menue_timeout.nil?
+      self.secure_boot = other.secure_boot unless other.secure_boot.nil?
+    end
+
+    def cpu_mitigations
+      Bootloader::CpuMitigations.new(:manual)
+    end
+
+    def cpu_mitigations=(value)
+      log.info "ignoring set of cpu_mitigations (#{value}) in systemd bootloader."
+    end
+
+    def read
+      super
+
+      read_menue_timeout
+      self.secure_boot = Systeminfo.secure_boot_active?
+    end
+
+    # Write bootloader settings to disk
+    def write(etc_only: false)
+      # super have to called as first as grub install require some config written in ancestor
+      super
+      log.info("Writing settings...")
+
+      install_bootloader
+      create_menue_entries
+      write_menue_timeout
+
+      true
+    end
+
+    def propose
+      super
+      log.info("Propose settings...")
+    end
+
+    def status_string(status)
+      if status
+        _("enabled")
+      else
+        _("disabled")
+      end
+    end
+
+    # Secure boot setting shown in summary screen.
+    #
+    # @return [String]
+    def secure_boot_summary
+      link = if secure_boot
+        "<a href=\"disable_secure_boot\">(#{_("disable")})</a>"
+      else
+        "<a href=\"enable_secure_boot\">(#{_("enable")})</a>"
+      end
+
+      "#{_("Secure Boot:")} #{status_string(secure_boot)} #{link}"
+    end
+
+    # Display bootloader summary
+    # @return a list of summary lines
+    def summary(*)
+      result = [
+        Yast::Builtins.sformat(
+          _("Boot Loader Type: %1"),
+          "Systemd Boot"
+        )
+      ]
+      result << secure_boot_summary if Systeminfo.secure_boot_available?(name)
+      result
+    end
+
+    def name
+      "systemd-boot"
+    end
+
+    def packages
+      res = super
+
+      case Yast::Arch.architecture
+      when "x86_64", "aarch64"
+        res << "shim" << "mokutil" if secure_boot
```

Comment:
so shim can work with systemd boot? and what about risc64? it also support EFI see https://github.com/yast/yast-bootloader/blob/master/src/lib/bootloader/systeminfo.rb#L98

Reply:
And also does it mean that systemd does not need any software for its installation? where lives bootctl?

Reply:
bootctl is currently located in udev (sub-package of systemd).
Just as a little warning, in Factory we are planning to move everything systemd-boot related into a new sub-package, for better separation soonish.

Reply:
ok, so when it happens, we will need to add such package here, so we are sure it is installed.

Reply:
I have restricted it to x86_64 only

Reply:
JFYI, since sr#1096633, bootctl (and everything else related to sd-boot) has been moved to a new subpackage systemd-boot now